base/strings/string.jl

```diff
@@ -104,6 +104,27 @@ function prevind(s::String, i::Integer)
     @inbounds while j > 0 && is_valid_continuation(codeunit(s,j))
         j -= 1
     end
+
+    j
+end
+
+function prevind(s::String, i::Integer, nchar::Integer)
+    nchar > 0 || throw(ArgumentError("nchar must be greater than 0"))
+    j = Int(i)
+    e = sizeof(s)
+
+    while nchar > 0
+        if j > e
+            j = endof(s)
+        else
+            j -= 1
+            @inbounds while j > 0 && is_valid_continuation(codeunit(s,j))
+                j -= 1
+            end
+        end
+        j <= 0 && return j
```

Comment:
This check wasn't present in the original code. Would there be a way to avoid it?

Reply:
this check is needed to make sure that `nextind(s,i,1)==nextind(s,i)` contract always holds.

Reply:
Yeah, but I wondered whether there would be a way to avoid it by adapting the code logic. Maybe not.

Reply:
I do not see one (and this is cheap).